salesforce/salesforce-inbound-endpoint/salesforce-inbound-endpoint-1.0.0/src/main/java/org/wso2/carbon/inbound/salesforce/poll/SoapLoginUtil.java

```diff
@@ -0,0 +1,100 @@
+ *  under the License.
+ */
+package org.wso2.carbon.inbound.salesforce.poll;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.eclipse.jetty.client.ContentExchange;
+import org.eclipse.jetty.client.HttpClient;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public final class SoapLoginUtil {
+    private static final Log log = LogFactory.getLog(SoapLoginUtil.class);
+
+    // The enterprise SOAP API endpoint used for the login call in this example.
+    private static final String SERVICES_SOAP_PARTNER_ENDPOINT = "/services/Soap/u/" + SalesforceConstant.SOAP_API_VERSION;
+
+    private static final String ENV_START =
+            "<soapenv:Envelope xmlns:soapenv='http://schemas.xmlsoap.org/soap/envelope/' "
+                    + "xmlns:xsi='http://www.w3.org/2001/XMLSchema-instance' " +
+                    "xmlns:urn='urn:partner.soap.sforce.com'><soapenv:Body>";
+
+    private static final String ENV_END = "</soapenv:Body></soapenv:Envelope>";
+    private static String sessionId, LoginUrl;
+
+    private static byte[] soapXmlForLogin(String username, String password)
+            throws UnsupportedEncodingException {
+        return (ENV_START +
+                "  <urn:login>" +
+                "    <urn:username>" + username + "</urn:username>" +
+                "    <urn:password>" + password + "</urn:password>" +
+                "  </urn:login>" +
+                ENV_END).getBytes("UTF-8");
+    }
+
+    public static void login(HttpClient client, String username, String password) throws IOException, InterruptedException, SAXException,
+            ParserConfigurationException {
+        try {
+            ContentExchange exchange = new ContentExchange();
+            exchange.setMethod("POST");
+            exchange.setURL(getSoapURL());
+            exchange.setRequestContentSource(new ByteArrayInputStream(soapXmlForLogin(
+                    username, password)));
+            exchange.setRequestHeader("Content-Type", "text/xml");
+            exchange.setRequestHeader("SOAPAction", "''");
+            exchange.setRequestHeader("PrettyPrint", "Yes");
+
+            client.send(exchange);
+            exchange.waitForDone();
+
+            String response = exchange.getResponseContent();
+            String tagSession = "<sessionId>";
+            String tagServerUrl = "<serverUrl>";
+            String serverUrl = response.substring(response.indexOf(tagServerUrl) + tagServerUrl.length(), response.indexOf("</serverUrl>"));
+            sessionId = response.substring(response.indexOf(tagSession) + tagSession.length(), response.indexOf("</sessionId>"));
+            LoginUrl = serverUrl.substring(0, serverUrl.indexOf("/services"));
```

Comment:
It will throw the exception like below

ERROR - SalesforceStreamData Error during make the client: String index out of range: -11
java.lang.StringIndexOutOfBoundsException: String index out of range: -11
	at java.lang.String.substring(String.java:1911)
	at org.wso2.carbon.inbound.salesforce.poll.SoapLoginUtil.login(SoapLoginUtil.java:66)
	at org.wso2.carbon.inbound.salesforce.poll.SalesforceStreamData.makeClient(SalesforceStreamData.java:177)
	at org.wso2.carbon.inbound.salesforce.poll.SalesforceStreamData.makeConnect(SalesforceStreamData.java:91)
	at org.wso2.carbon.inbound.salesforce.poll.SalesforceStreamData.poll(SalesforceStreamData.java:288)
	at org.wso2.carbon.inbound.endpoint.protocol.generic.GenericTask.taskExecute(GenericTask.java:39)
	at org.wso2.carbon.inbound.endpoint.common.InboundTask.execute(InboundTask.java:50)
	at org.wso2.carbon.mediation.ntask.NTaskAdapter.execute(NTaskAdapter.java:90)
	at org.wso2.carbon.ntask.core.impl.TaskQuartzJobAdapter.execute(TaskQuartzJobAdapter.java:67)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
[2015-12-17 12:22:58,748] ERROR - SalesforceStreamData Error while setup the Salesforce connection.
org.apache.synapse.SynapseException: java.lang.StringIndexOutOfBoundsException: String index out of range: -11
	at org.wso2.carbon.inbound.salesforce.poll.SalesforceStreamData.handleException(SalesforceStreamData.java:315)
	at org.wso2.carbon.inbound.salesforce.poll.SalesforceStreamData.makeConnect(SalesforceStreamData.java:93)
	at org.wso2.carbon.inbound.salesforce.poll.SalesforceStreamData.poll(SalesforceStreamData.java:288)
	at org.wso2.carbon.inbound.endpoint.protocol.generic.GenericTask.taskExecute(GenericTask.java:39)
	at org.wso2.carbon.inbound.endpoint.common.InboundTask.execute(InboundTask.java:50)
	at org.wso2.carbon.mediation.ntask.NTaskAdapter.execute(NTaskAdapter.java:90)
	at org.wso2.carbon.ntask.core.impl.TaskQuartzJobAdapter.execute(TaskQuartzJobAdapter.java:67)
	at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: -11
	at java.lang.String.substring(String.java:1911)
	at org.wso2.carbon.inbound.salesforce.poll.SoapLoginUtil.login(SoapLoginUtil.java:66)
	at org.wso2.carbon.inbound.salesforce.poll.SalesforceStreamData.makeClient(SalesforceStreamData.java:177)
	at org.wso2.carbon.inbound.salesforce.poll.SalesforceStreamData.makeConnect(SalesforceStreamData.java:91)
	... 11 more